subsys/pm/pm.c

```diff
@@ -203,7 +203,7 @@ bool pm_system_suspend(int32_t ticks)
 
 	SYS_PORT_TRACING_FUNC_ENTER(pm, system_suspend, ticks);
 
-	if (!atomic_test_and_set_bit(z_cpus_pm_state_forced, id)) {
+	if (!atomic_test_bit(z_cpus_pm_state_forced, id)) {
```

Comment:
What about `atomic_test_and_clear_bit` ?

Reply:
Nope, it has to continue set until be used otherwise another thread can force a different state before it be used.